---
title: "Differential Abundance Analysis (DAA): Fecal microbiome"
author: "Eva Kim"
output: 
  html_document:
    df_print: paged
    toc: true
    toc_float: true
    toc_collapsed: true
    toc_depth: 3
    number_sections: true
    theme: lumen
date: "`r format(Sys.time(), '%m/%d/%Y')`"
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

Based on the alpha and beta diversity results, I am not expecting to find a huge difference between groups in terms of bacterial composition. Please note that many of these code are from the Microbiome Intensive Course (MIC Course, Duke University, 2023) and thus the work of [Dr. Pixu Shi](https://scholars.duke.edu/person/Pixu.Shi). 

This is after talking with Dr. Shi about what analyses I want to run. 

# Set-Up

First, update the `ANCOMBC` package if necessary. 

```{r, results = 'hide', warning=FALSE, message=FALSE}
#if (!requireNamespace("BiocManager", quietly=TRUE))
#     install.packages("BiocManager")
#BiocManager::install("ANCOMBC")
```


# Load libraries and directories

```{r, results = 'hide', warning=FALSE, message=FALSE}
# rm(list = ls(all = TRUE)) # Unload all packages if necessary
library(phyloseq)
library(fs)
library(rstatix)
library(ggpubr)
library(microViz)
library(ANCOMBC)
library(pals)
library(RColorBrewer)
library(colorBlindness)
library(microbiome)
library(ggrepel)
library(DT)
library(plotly)
library(DESeq2)
library(parallel)
library(dplyr)
library(tidyr)
library(readr)
library(tibble)
library(stringr)
library(cowplot)
```


```{r, warning=FALSE, message=FALSE}
rm(list=ls()) # clear environment 

git.dir = file.path("/hpc/group/dmc/Eva/GLP_KO_Microbiome_final") # CHANGE ME 
ps.rds <- file.path(git.dir, "ps.GLP1.rds")
ps <- read_rds(ps.rds)
ps

fig.dir = file.path(git.dir, "figures")

map.file = file.path(git.dir, "Metadata_GLP1.csv")
meta.df = read_csv(map.file, show_col_types = FALSE)
```

# Prepare phyloseq object


I will now aggregate at the genus level, as it is the smallest taxonomic rank that analyses will be run on. In other words, I do not plan on running differential abundance analysis at ASV level. 

Let's prepare the phyloseq object for differential abundance analysis (DAA). Since we are interested in 1) differentially abundant taxa and 2) correlation analysis, we will limit the taxa to only ones that appear in > 3 samples for *each sample type*. 

To avoid running analyses on extremely rare taxa that may be sequencing artifacts, the threshold for minimum number of counts is set at 10. This is, of course, arbitrary. 10 was chosen as 1% of number of read threshold of 1000 above for pruning samples. Based on how I wrote the code, taxa must be present have > 10 reads or more in > 3 samples. 

Furthermore, categorical variables should be re-leveled as factors.

```{r, warning=FALSE, message=FALSE}
# Subset to true fecal samples only and remove tree
ps  %>% subset_samples(Sample_type == "Fecal" & Type == "True Sample") -> ps_fecal
ps_fecal_notree <- phyloseq(otu_table(ps_fecal), tax_table(ps_fecal), sample_data(ps_fecal))

# Categorical variables should be factors 
ps_fecal_notree@sam_data$Genotype <- factor(ps_fecal_notree@sam_data$Genotype, levels = c("WT", "KO"))
ps_fecal_notree@sam_data$Sex <- factor(ps_fecal_notree@sam_data$Sex, levels = c("Female", "Male"))
ps_fecal_notree@sam_data$Intranasal_Treatment <- factor(ps_fecal_notree@sam_data$Intranasal_Treatment, levels = c("PBS", "HDM"))
ps_fecal_notree@sam_data$Surgery <- factor(ps_fecal_notree@sam_data$Surgery, levels = c("None", "Sham", "VSG"))
ps_fecal_notree@sam_data$Mouse %>% unique() -> mouse_levels
ps_fecal_notree@sam_data$Mouse <- factor(ps_fecal_notree@sam_data$Mouse, levels = mouse_levels)
ps_fecal_notree@sam_data$Diet <- factor(ps_fecal_notree@sam_data$Diet, levels = c("Normal_Chow", "High_Fat_Diet"))
ps_fecal_notree@sam_data$Group <- factor(ps_fecal_notree@sam_data$Group, levels = c("NA", "Control", "1", "2"))
ps_fecal_notree@sam_data$Week <- factor(ps_fecal_notree@sam_data$Week, levels = c("0", "10", "13"))
ps_fecal_notree

# Aggregate at genus level
ps_g_notree_fecal <- tax_glom(ps_fecal_notree, "Genus")
taxa_names(ps_g_notree_fecal) <- tax_table(ps_g_notree_fecal)[, 'Genus']

ps_g_notree_fecal
```

# Getting ready: prune data

We will prune the phyloseq object at each phylogenetic rank at and above genus. Since we plan on running correlation analysis after differential abundance, let's test for taxa that are present in at least 5 samples (arbitrary number) with greater than 10 reads in each sample (also arbitrary. It's also 1% of 1000 reads, which was the threshold used to remove samples from analysis). 

x` Genus 

```{r}
# Pruning: set threshold number 
threshold_num = 4
nreads_prune = 10

# Prune taxa at each phylogenetic rank 
ps_g_notree_fecal %>%
  filter_taxa(., function(x) {sum(x > nreads_prune) > threshold_num}, prune = TRUE) -> ps_genus.prune
ps_genus.prune

ps_g_notree_fecal %>% transform_sample_counts(function(x) x/sum(x)) -> ps_genus.ts
ps_genus.ts

subset_taxa(ps_genus.ts, ps_genus.ts@tax_table[, "Genus"] %in% ps_genus.prune@tax_table[, "Genus"]) -> ps_genus.prune.ts
ps_genus.prune.ts

rowSums(ps_genus.prune.ts@otu_table) %>% as.data.frame() %>% 
  dplyr::rename(RelAb = ".") %>% mutate(RelAb = RelAb * 100,
                                        RelAb = format(RelAb, scientific = FALSE),
                                        RelAb = as.numeric(RelAb)) %>% 
  dplyr::arrange(RelAb) -> genus_prune_RelAb

summary(genus_prune_RelAb$RelAb)
```


## Family 
 
```{r}
ps_fecal_notree %>% tax_glom("Family") -> ps_Family  
taxa_names(ps_Family) <- tax_table(ps_Family)[, 'Family']
ps_Family

ps_Family %>% transform_sample_counts(function(x) x/sum(x)) -> ps_Family.ts
ps_Family.ts

# Prune taxa at each phylogenetic rank 
ps_Family %>%
  filter_taxa(., function(x) {sum(x > nreads_prune) > threshold_num}, prune = TRUE) -> ps_Family.prune
ps_Family.prune

subset_taxa(ps_Family.ts, ps_Family.ts@tax_table[, "Family"] %in% ps_Family.prune@tax_table[, "Family"]) -> ps_Family.prune.ts
ps_Family.prune.ts

rowSums(ps_Family.prune.ts@otu_table) %>% as.data.frame() %>% 
  dplyr::rename(RelAb = ".") %>% mutate(RelAb = RelAb * 100,
                                        RelAb = format(RelAb, scientific = FALSE),
                                        RelAb = as.numeric(RelAb)) %>% 
  dplyr::arrange(RelAb) -> Family_prune_RelAb

summary(Family_prune_RelAb)
```

## Order 
 
```{r}
ps_fecal_notree %>% tax_glom("Order") -> ps_Order  
taxa_names(ps_Order) <- tax_table(ps_Order)[, 'Order']
ps_Order

ps_Order %>% transform_sample_counts(function(x) x/sum(x)) -> ps_Order.ts
ps_Order.ts

# Prune taxa at each phylogenetic rank 
ps_Order %>%
  filter_taxa(., function(x) {sum(x > nreads_prune) > threshold_num}, prune = TRUE) -> ps_Order.prune
ps_Order.prune

subset_taxa(ps_Order.ts, ps_Order.ts@tax_table[, "Order"] %in% ps_Order.prune@tax_table[, "Order"]) -> ps_Order.prune.ts
ps_Order.prune.ts

rowSums(ps_Order.prune.ts@otu_table) %>% as.data.frame() %>% 
  dplyr::rename(RelAb = ".") %>% mutate(RelAb = RelAb * 100,
                                        RelAb = format(RelAb, scientific = FALSE),
                                        RelAb = as.numeric(RelAb)) %>% 
  dplyr::arrange(RelAb) -> Order_prune_RelAb

summary(Order_prune_RelAb)
```

## Class 
 
```{r}
ps_fecal_notree %>% tax_glom("Class") -> ps_Class  
taxa_names(ps_Class) <- tax_table(ps_Class)[, 'Class']
ps_Class

ps_Class %>% transform_sample_counts(function(x) x/sum(x)) -> ps_Class.ts
ps_Class.ts

# Prune taxa at each phylogenetic rank 
ps_Class %>%
  filter_taxa(., function(x) {sum(x > nreads_prune) > threshold_num}, prune = TRUE) -> ps_Class.prune
ps_Class.prune

subset_taxa(ps_Class.ts, ps_Class.ts@tax_table[, "Class"] %in% ps_Class.prune@tax_table[, "Class"]) -> ps_Class.prune.ts
ps_Class.prune.ts

rowSums(ps_Class.prune.ts@otu_table) %>% as.data.frame() %>% 
  dplyr::rename(RelAb = ".") %>% mutate(RelAb = RelAb * 100,
                                        RelAb = format(RelAb, scientific = FALSE),
                                        RelAb = as.numeric(RelAb)) %>% 
  dplyr::arrange(RelAb) -> Class_prune_RelAb

summary(Class_prune_RelAb)
```

## Phylum 
 
```{r}
ps_fecal_notree %>% tax_glom("Phylum") -> ps_Phylum  
taxa_names(ps_Phylum) <- tax_table(ps_Phylum)[, 'Phylum']
ps_Phylum

ps_Phylum %>% transform_sample_counts(function(x) x/sum(x)) -> ps_Phylum.ts
ps_Phylum.ts

# Prune taxa at each phylogenetic rank 
ps_Phylum %>%
  filter_taxa(., function(x) {sum(x > nreads_prune) > threshold_num}, prune = TRUE) -> ps_Phylum.prune
ps_Phylum.prune

subset_taxa(ps_Phylum.ts, ps_Phylum.ts@tax_table[, "Phylum"] %in% ps_Phylum.prune@tax_table[, "Phylum"]) -> ps_Phylum.prune.ts
ps_Phylum.prune.ts

rowSums(ps_Phylum.prune.ts@otu_table) %>% as.data.frame() %>% 
  dplyr::rename(RelAb = ".") %>% mutate(RelAb = RelAb * 100,
                                        RelAb = format(RelAb, scientific = FALSE),
                                        RelAb = as.numeric(RelAb)) %>% 
  dplyr::arrange(RelAb) -> Phylum_prune_RelAb

summary(Phylum_prune_RelAb)
```

# Visualization

Let's visualize at Phylum level to see how much information was kept: 

```{r}
plot_bar(ps_genus.prune.ts, x = "Label", fill="Phylum") + 
  facet_grid(scales="free", space = "free_x") + 
  geom_bar(aes(color=Phylum, fill=Phylum), stat="identity", position="stack") +  
  theme_bw()  +   
  coord_cartesian(ylim = c(0,1), expand=0) +
  theme(axis.text.x=element_blank()) + # not showing samples names 
  labs(y="Relative Abundance", title = "Relative Abundance of fecal Samples, Pruned @Genus", x = "Sample")+
  scale_color_manual(values=SteppedSequential5Steps) + 
  scale_fill_manual(values=SteppedSequential5Steps) 
```

```{r}
plot_bar(ps_Family.prune.ts, x = "Label", fill="Phylum") + 
  facet_grid(scales="free", space = "free_x") + 
  geom_bar(aes(color=Phylum, fill=Phylum), stat="identity", position="stack") +  
  theme_bw()  +   
  coord_cartesian(ylim = c(0,1), expand=0) +
  theme(axis.text.x=element_blank()) + # not showing samples names 
  labs(y="Relative Abundance", title = "Relative Abundance of fecal Samples, Pruned @Family", x = "Sample")+
  scale_color_manual(values=SteppedSequential5Steps) + 
  scale_fill_manual(values=SteppedSequential5Steps) 
```

```{r}
plot_bar(ps_Order.prune.ts, x = "Label", fill="Phylum") + 
  facet_grid(scales="free", space = "free_x") + 
  geom_bar(aes(color=Phylum, fill=Phylum), stat="identity", position="stack") +  
  theme_bw()  +   
  coord_cartesian(ylim = c(0,1), expand=0) +
  theme(axis.text.x=element_blank()) + # not showing samples names 
  labs(y="Relative Abundance", title = "Relative Abundance of fecal Samples, Pruned @Order", x = "Sample")+
  scale_color_manual(values=SteppedSequential5Steps) + 
  scale_fill_manual(values=SteppedSequential5Steps) 
```

```{r}
plot_bar(ps_Class.prune.ts, x = "Label", fill="Class") + 
  facet_grid(scales="free", space = "free_x") + 
  geom_bar(aes(color=Class, fill=Class), stat="identity", position="stack") +  
  theme_bw()  +   
  coord_cartesian(ylim = c(0,1), expand=0) +
  theme(axis.text.x=element_blank()) + # not showing samples names 
  labs(y="Relative Abundance", title = "Relative Abundance of fecal Samples, Pruned @Class", x = "Sample")+
  scale_color_manual(values=SteppedSequential5Steps) + 
  scale_fill_manual(values=SteppedSequential5Steps) 
```

```{r}
plot_bar(ps_Phylum.prune.ts, x = "Label", fill="Phylum") + 
  facet_grid(scales="free", space = "free_x") + 
  geom_bar(aes(color=Phylum, fill=Phylum), stat="identity", position="stack") +  
  theme_bw()  +   
  coord_cartesian(ylim = c(0,1), expand=0) +
  theme(axis.text.x=element_blank()) + # not showing samples names 
  labs(y="Relative Abundance", title = "Relative Abundance of fecal Samples, Pruned @Phylum", x = "Sample")+
  scale_color_manual(values=SteppedSequential5Steps) + 
  scale_fill_manual(values=SteppedSequential5Steps) 
```

# Subset metadata 

```{r}
meta.df %>% filter(Type == "True Sample" & Sample_type == "Fecal") %>% 
  filter(Label %in% phyloseq::sample_data(ps_g_notree_fecal)$Label) -> meta.fecal
meta.fecal
```



# Q1: Fecal microbiome, HFD, wk 10 & 13,  Surgery * Week

Let's see how many mice we have: 

```{r}
meta.fecal %>% filter(Diet == "High_Fat_Diet" & Week != 0) -> meta.f.hfd
meta.f.hfd %>% 
  dplyr::count(Week, Surgery, Genotype) %>% 
  arrange(n)
```

```{r}
meta.f.hfd %>% 
  dplyr::count(Week, Surgery) %>% 
  arrange(n)
```


```{r}
nrow(meta.f.hfd)
```

```{r}
meta.f.hfd %>% distinct(Mouse) %>% nrow()
```


## Make TSE objects

```{r}
ps_genus.prune %>% subset_samples(Diet == "High_Fat_Diet" & Week !="0")  %>%
  mia::makeTreeSummarizedExperimentFromPhyloseq() -> tse.fecal.Q1.Genus
tse.fecal.Q1.Genus
```

```{r}
ps_Family.prune %>% subset_samples(Diet == "High_Fat_Diet" & Week !="0")  %>%
  mia::makeTreeSummarizedExperimentFromPhyloseq() -> tse.fecal.Q1.Family
tse.fecal.Q1.Family

ps_Order.prune %>% subset_samples(Diet == "High_Fat_Diet" & Week !="0")  %>%
  mia::makeTreeSummarizedExperimentFromPhyloseq() -> tse.fecal.Q1.Order
tse.fecal.Q1.Order

ps_Class.prune %>% subset_samples(Diet == "High_Fat_Diet" & Week !="0")  %>%
  mia::makeTreeSummarizedExperimentFromPhyloseq() -> tse.fecal.Q1.Class
tse.fecal.Q1.Class

ps_Phylum.prune %>% subset_samples(Diet == "High_Fat_Diet" & Week !="0")  %>%
  mia::makeTreeSummarizedExperimentFromPhyloseq() -> tse.fecal.Q1.Phylum
tse.fecal.Q1.Phylum
```

## ANCOM2

### Genus

```{r}
set.seed(123) # Set seed for reproducibility 
n_cl_requested = 24

# Genus level
fecal.Q1.Genus <- ANCOMBC::ancom(data=tse.fecal.Q1.Genus, assay_name="counts", tax_level="Genus",
                               p_adj_method="holm", prv_cut=0, lib_cut=0, 
                               main_var="Surgery", adj_formula="Week + Genotype",
                               alpha = 0.05, struc_zero=TRUE,
                               n_cl = n_cl_requested) 

res.fecal.Q1.Genus = fecal.Q1.Genus$res
res.fecal.Q1.Genus %>% filter(detected_0.7 == TRUE) -> tab.fecal.Q1.Genus.sig
tab.fecal.Q1.Genus.sig 
```

## Family

```{r}
set.seed(123) # Set seed for reproducibility 

# Family level
fecal.Q1.Family <- ANCOMBC::ancom(data=tse.fecal.Q1.Family, assay_name="counts", tax_level="Family",
                               p_adj_method="holm", prv_cut=0, lib_cut=0, 
                               main_var="Surgery", adj_formula="Week + Genotype",
                               alpha = 0.05, struc_zero=TRUE,
                               n_cl = n_cl_requested) 

res.fecal.Q1.Family = fecal.Q1.Family$res
res.fecal.Q1.Family %>% filter(detected_0.7 == TRUE) -> tab.fecal.Q1.Family.sig
tab.fecal.Q1.Family.sig 
```

### Order

```{r}
set.seed(123) # Set seed for reproducibility 

# Order level
fecal.Q1.Order <- ANCOMBC::ancom(data=tse.fecal.Q1.Order, assay_name="counts", tax_level="Order",
                               p_adj_method="holm", prv_cut=0, lib_cut=0, 
                               main_var="Surgery", adj_formula="Week + Genotype",
                               alpha = 0.05, struc_zero=TRUE,
                               n_cl = n_cl_requested) 

res.fecal.Q1.Order = fecal.Q1.Order$res
res.fecal.Q1.Order %>% filter(detected_0.7 == TRUE) -> tab.fecal.Q1.Order.sig
tab.fecal.Q1.Order.sig 
```

### Class

```{r}
set.seed(123) # Set seed for reproducibility 

# Class level
fecal.Q1.Class <- ANCOMBC::ancom(data=tse.fecal.Q1.Class, assay_name="counts", tax_level="Class",
                               p_adj_method="holm", prv_cut=0, lib_cut=0, 
                               main_var="Surgery", adj_formula="Week + Genotype",
                               alpha = 0.05, struc_zero=TRUE,
                               n_cl = n_cl_requested) 

res.fecal.Q1.Class = fecal.Q1.Class$res
res.fecal.Q1.Class %>% filter(detected_0.7 == TRUE) -> tab.fecal.Q1.Class.sig
tab.fecal.Q1.Class.sig 
```

### Phylum

```{r}
set.seed(123) # Set seed for reproducibility 

# Phylum level
fecal.Q1.Phylum <- ANCOMBC::ancom(data=tse.fecal.Q1.Phylum, assay_name="counts", tax_level="Phylum",
                               p_adj_method="holm", prv_cut=0, lib_cut=0, 
                               main_var="Surgery", adj_formula="Week + Genotype",
                               alpha = 0.05, struc_zero=TRUE,
                               n_cl = n_cl_requested) 

res.fecal.Q1.Phylum = fecal.Q1.Phylum$res
res.fecal.Q1.Phylum %>% filter(detected_0.7 == TRUE) -> tab.fecal.Q1.Phylum.sig
tab.fecal.Q1.Phylum.sig 
```

### Aggregate Results

```{r}
tab.fecal.Q1.Genus.sig$Tax_rank <- "Genus"
tab.fecal.Q1.Family.sig$Tax_rank <- "Family"
tab.fecal.Q1.Order.sig$Tax_rank <- "Order"
tab.fecal.Q1.Class.sig$Tax_rank <- "Class"
tab.fecal.Q1.Phylum.sig$Tax_rank <- "Phylum"

rbind(tab.fecal.Q1.Genus.sig,
      tab.fecal.Q1.Family.sig,
      tab.fecal.Q1.Order.sig,
      tab.fecal.Q1.Class.sig,
      tab.fecal.Q1.Phylum.sig) -> tab.fecal.Q1

tab.fecal.Q1$Test <- "ANCOM II, fecal microbiome: all HFD, wk 10 & 13. Structural zeroes from Surgery, adjust with Week + Genotype"

tab.fecal.Q1 
```

## DESeq2

###  Genus 

```{r}
ds.fecal.Q1_Genus <- DESeq2::DESeqDataSet(tse.fecal.Q1.Genus, ~ Surgery * Week + Genotype)
dds.fecal.Q1_Genus <- DESeq2::DESeq(ds.fecal.Q1_Genus)

DESeq2::summary(dds.fecal.Q1_Genus)
resultsNames(dds.fecal.Q1_Genus)
```

```{r}
results(dds.fecal.Q1_Genus, tidy = TRUE, contrast = c("Week", "13", "10")) %>% # tidy converts to df
  filter(padj < 0.05) -> fecal.Q1.Genus.Res
fecal.Q1.Genus.Res$Comparison <- c("Week: week 13 over week 10")

results(dds.fecal.Q1_Genus, contrast = c("Surgery", "Sham", "None"), tidy = TRUE) %>%
  filter(padj < 0.05) -> tmp
tmp$Comparison <- c("Surgery: Sham over None")
rbind(tmp, fecal.Q1.Genus.Res) -> fecal.Q1.Genus.Res

results(dds.fecal.Q1_Genus, contrast = c("Surgery", "VSG", "None"), tidy = TRUE) %>%
  filter(padj < 0.05) -> tmp
tmp$Comparison <- c("Surgery: VSG over None")
rbind(tmp, fecal.Q1.Genus.Res) -> fecal.Q1.Genus.Res

results(dds.fecal.Q1_Genus, name  = c("SurgerySham.Week13"), tidy = TRUE) %>%
  filter(padj < 0.05) -> tmp
tmp$Comparison <- c("Surgery Sham, wk 13")
rbind(tmp, fecal.Q1.Genus.Res) -> fecal.Q1.Genus.Res

results(dds.fecal.Q1_Genus, name  = c("SurgeryVSG.Week13"), tidy = TRUE) %>%
  filter(padj < 0.05) -> tmp
tmp$Comparison <- c("Surgery VSG, wk 13")
rbind(tmp, fecal.Q1.Genus.Res) -> fecal.Q1.Genus.Res

results(dds.fecal.Q1_Genus, name  = c("Genotype_KO_vs_WT"), tidy = TRUE) %>%
  filter(padj < 0.05) -> tmp
tmp$Comparison <- c("Surgery VSG, wk 13")
rbind(tmp, fecal.Q1.Genus.Res) -> fecal.Q1.Genus.Res

fecal.Q1.Genus.Res$Compare_taxon <- "Genus"

fecal.Q1.Genus.Res
```

```{r}
fecal.Q1.Genus.Res %>% filter(row %in% tab.fecal.Q1.Genus.sig)
```

### Family 

```{r}
ds.fecal.Q1_Family <- DESeq2::DESeqDataSet(tse.fecal.Q1.Family, ~ Surgery * Week + Genotype)
dds.fecal.Q1_Family <- DESeq2::DESeq(ds.fecal.Q1_Family)

DESeq2::summary(dds.fecal.Q1_Family)
resultsNames(dds.fecal.Q1_Family)
```

```{r}
results(dds.fecal.Q1_Family, tidy = TRUE, contrast = c("Week", "13", "10")) %>% # tidy converts to df
  filter(padj < 0.05) -> fecal.Q1.Family.Res
fecal.Q1.Family.Res$Comparison <- c("Week: week 13 over week 10")

results(dds.fecal.Q1_Family, contrast = c("Surgery", "Sham", "None"), tidy = TRUE) %>%
  filter(padj < 0.05) -> tmp
tmp$Comparison <- c("Surgery: Sham over None")
rbind(tmp, fecal.Q1.Family.Res) -> fecal.Q1.Family.Res

results(dds.fecal.Q1_Family, contrast = c("Surgery", "VSG", "None"), tidy = TRUE) %>%
  filter(padj < 0.05) -> tmp
tmp$Comparison <- c("Surgery: VSG over None")
rbind(tmp, fecal.Q1.Family.Res) -> fecal.Q1.Family.Res

results(dds.fecal.Q1_Family, name  = c("SurgerySham.Week13"), tidy = TRUE) %>%
  filter(padj < 0.05) -> tmp
tmp$Comparison <- c("Surgery Sham, wk 13")
rbind(tmp, fecal.Q1.Family.Res) -> fecal.Q1.Family.Res

results(dds.fecal.Q1_Family, name  = c("SurgeryVSG.Week13"), tidy = TRUE) %>%
  filter(padj < 0.05) -> tmp
tmp$Comparison <- c("Surgery VSG, wk 13")
rbind(tmp, fecal.Q1.Family.Res) -> fecal.Q1.Family.Res

results(dds.fecal.Q1_Family, name  = c("Genotype_KO_vs_WT"), tidy = TRUE) %>%
  filter(padj < 0.05) -> tmp
tmp$Comparison <- c("Surgery VSG, wk 13")
rbind(tmp, fecal.Q1.Family.Res) -> fecal.Q1.Family.Res

fecal.Q1.Family.Res$Compare_taxon <- "Family"

fecal.Q1.Family.Res
```

```{r}
fecal.Q1.Family.Res %>% filter(row %in% tab.fecal.Q1.Family.sig$taxon)
```


### Order 

```{r}
ds.fecal.Q1_Order <- DESeq2::DESeqDataSet(tse.fecal.Q1.Order, ~ Surgery * Week  + Genotype)
dds.fecal.Q1_Order <- DESeq2::DESeq(ds.fecal.Q1_Order)

DESeq2::summary(dds.fecal.Q1_Order)
resultsNames(dds.fecal.Q1_Order)
```

```{r}
results(dds.fecal.Q1_Order, tidy = TRUE, contrast = c("Week", "13", "10")) %>% # tidy converts to df
  filter(padj < 0.05) -> fecal.Q1.Order.Res
fecal.Q1.Order.Res$Comparison <- c("Week: week 13 over week 10")

results(dds.fecal.Q1_Order, contrast = c("Surgery", "Sham", "None"), tidy = TRUE) %>%
  filter(padj < 0.05) -> tmp
tmp$Comparison <- c("Surgery: Sham over None")
rbind(tmp, fecal.Q1.Order.Res) -> fecal.Q1.Order.Res

results(dds.fecal.Q1_Order, contrast = c("Surgery", "VSG", "None"), tidy = TRUE) %>%
  filter(padj < 0.05) -> tmp
tmp$Comparison <- c("Surgery: VSG over None")
rbind(tmp, fecal.Q1.Order.Res) -> fecal.Q1.Order.Res

results(dds.fecal.Q1_Order, name  = c("SurgerySham.Week13"), tidy = TRUE) %>%
  filter(padj < 0.05) -> tmp
tmp$Comparison <- c("Surgery Sham, wk 13")
rbind(tmp, fecal.Q1.Order.Res) -> fecal.Q1.Order.Res

results(dds.fecal.Q1_Order, name  = c("SurgeryVSG.Week13"), tidy = TRUE) %>%
  filter(padj < 0.05) -> tmp
tmp$Comparison <- c("Surgery VSG, wk 13")
rbind(tmp, fecal.Q1.Order.Res) -> fecal.Q1.Order.Res

results(dds.fecal.Q1_Order, name  = c("Genotype_KO_vs_WT"), tidy = TRUE) %>%
  filter(padj < 0.05) -> tmp
tmp$Comparison <- c("Surgery VSG, wk 13")
rbind(tmp, fecal.Q1.Order.Res) -> fecal.Q1.Order.Res

fecal.Q1.Order.Res$Compare_taxon <- "Order"

fecal.Q1.Order.Res
fecal.Q1.Order.Res %>% filter(row %in% tab.fecal.Q1.Order.sig$taxon)
```


### Class 

```{r}
ds.fecal.Q1_Class <- DESeq2::DESeqDataSet(tse.fecal.Q1.Class, ~ Surgery * Week  + Genotype)
dds.fecal.Q1_Class <- DESeq2::DESeq(ds.fecal.Q1_Class)

DESeq2::summary(dds.fecal.Q1_Class)
resultsNames(dds.fecal.Q1_Class)
```

```{r}
results(dds.fecal.Q1_Class, tidy = TRUE, contrast = c("Week", "13", "10")) %>% # tidy converts to df
  filter(padj < 0.05) -> fecal.Q1.Class.Res
fecal.Q1.Class.Res$Comparison <- c("Week: week 13 over week 10")

results(dds.fecal.Q1_Class, contrast = c("Surgery", "Sham", "None"), tidy = TRUE) %>%
  filter(padj < 0.05) 

results(dds.fecal.Q1_Class, contrast = c("Surgery", "VSG", "None"), tidy = TRUE) %>%
  filter(padj < 0.05) -> tmp
tmp$Comparison <- c("Surgery: VSG over None")
rbind(tmp, fecal.Q1.Class.Res) -> fecal.Q1.Class.Res

results(dds.fecal.Q1_Class, name  = c("SurgerySham.Week13"), tidy = TRUE) %>%
  filter(padj < 0.05) 

results(dds.fecal.Q1_Class, name  = c("SurgeryVSG.Week13"), tidy = TRUE) %>%
  filter(padj < 0.05) -> tmp
tmp$Comparison <- c("Surgery VSG, wk 13")
rbind(tmp, fecal.Q1.Class.Res) -> fecal.Q1.Class.Res

results(dds.fecal.Q1_Class, name  = c("Genotype_KO_vs_WT"), tidy = TRUE) %>%
  filter(padj < 0.05) -> tmp
tmp$Comparison <- c("Surgery VSG, wk 13")
rbind(tmp, fecal.Q1.Class.Res) -> fecal.Q1.Class.Res

fecal.Q1.Class.Res$Compare_taxon <- "Class"

fecal.Q1.Class.Res
fecal.Q1.Class.Res %>% filter(row %in% tab.fecal.Q1.Class.sig$taxon)
```


### Phylum 

```{r}
ds.fecal.Q1_Phylum <- DESeq2::DESeqDataSet(tse.fecal.Q1.Phylum, ~ Surgery * Week  + Genotype)
dds.fecal.Q1_Phylum <- DESeq2::DESeq(ds.fecal.Q1_Phylum)

DESeq2::summary(dds.fecal.Q1_Phylum)
resultsNames(dds.fecal.Q1_Phylum)
```

```{r}
results(dds.fecal.Q1_Phylum, tidy = TRUE, contrast = c("Week", "13", "10")) %>% # tidy converts to df
  filter(padj < 0.05) -> fecal.Q1.Phylum.Res
fecal.Q1.Phylum.Res$Comparison <- c("Week: week 13 over week 10")

results(dds.fecal.Q1_Phylum, contrast = c("Surgery", "Sham", "None"), tidy = TRUE) %>%
  filter(padj < 0.05) 

results(dds.fecal.Q1_Phylum, contrast = c("Surgery", "VSG", "None"), tidy = TRUE) %>%
  filter(padj < 0.05) 

results(dds.fecal.Q1_Phylum, name  = c("SurgerySham.Week13"), tidy = TRUE) %>%
  filter(padj < 0.05) 

results(dds.fecal.Q1_Phylum, name  = c("SurgeryVSG.Week13"), tidy = TRUE) %>%
  filter(padj < 0.05) -> tmp
tmp$Comparison <- c("Surgery VSG, wk 13")
rbind(tmp, fecal.Q1.Phylum.Res) -> fecal.Q1.Phylum.Res

results(dds.fecal.Q1_Phylum, name  = c("Genotype_KO_vs_WT"), tidy = TRUE) %>%
  filter(padj < 0.05) -> tmp
tmp$Comparison <- c("Surgery VSG, wk 13")
rbind(tmp, fecal.Q1.Phylum.Res) -> fecal.Q1.Phylum.Res

fecal.Q1.Phylum.Res$Compare_taxon <- "Phylum"

fecal.Q1.Phylum.Res
fecal.Q1.Phylum.Res %>% filter(row %in% tab.fecal.Q1.Phylum.sig$taxon)
```

### Aggregate results

```{r}
rbind(fecal.Q1.Genus.Res, 
      fecal.Q1.Family.Res,
      fecal.Q1.Class.Res,
      fecal.Q1.Order.Res,
      fecal.Q1.Phylum.Res) -> fecal.Q1.deseq

fecal.Q1.deseq$Test <- "DESeq2, Fecal microbiome: all HFD, wk 10 + 13, Surgery * Week + Genotype"

fecal.Q1.deseq
```

## Intersection 

```{r}
tab.fecal.Q1 %>% filter(taxon %in% fecal.Q1.deseq$row) -> ancom.intersect.hfd
ancom.intersect.hfd
```

```{r}
fecal.Q1.deseq %>% filter(row %in% tab.fecal.Q1$taxon) 
```

## Graph

```{r}
psmelt(ps_genus.prune.ts) -> Genus.prune.melt 
psmelt(ps_Family.prune.ts) -> Family.prune.melt
psmelt(ps_Order.prune.ts) -> Order.prune.melt
psmelt(ps_Class.prune.ts) -> Class.prune.melt
psmelt(ps_Phylum.prune.ts) -> Phylum.prune.melt
```


```{r, warning=FALSE}
bind_rows(
  Genus.prune.melt,
  Family.prune.melt,
  Order.prune.melt,
  Class.prune.melt,
  Phylum.prune.melt
)  -> fecal.RA.Full

fecal.RA.Full$Surgery <- factor(fecal.RA.Full$Surgery, levels = c("None", "Sham", "VSG"))

fecal.RA.Full %>% 
  dplyr::select(OTU:Label, Sample_type:Mouse, Genotype:Surgery, Week) -> fecal.RA.Full

head(fecal.RA.Full)
tail(fecal.RA.Full)
```

```{r}
fecal.RA.Full %>% filter(Diet == "High_Fat_Diet" & Week != "0") -> fecal.graph.hfd
head(fecal.graph.hfd)
tail(fecal.graph.hfd)
```

```{r}
surgery.labs <- c("No Surgery", "Sham Surgery", "VSG")
names(surgery.labs) <- c("None", "Sham", "VSG")

fecal.graph.hfd %>% 
  filter(OTU %in% ancom.intersect.hfd$taxon) %>%   
  ggplot(aes(x = Surgery, y = Abundance, color = Surgery)) + 
  geom_boxplot() + geom_point() +  
  labs(y="Relative Abundance", title = str_glue(ancom.intersect.hfd$taxon[1], " Relative Abundance, fecal microbiome"),
       x="Surgery") + 
  theme_bw(base_size = 14) + 
  facet_wrap(~Week)

```

```{r}
fecal.graph.hfd %>% 
  filter(OTU %in% ancom.intersect.hfd$taxon) %>%   
  ggplot(aes(x = Week, y = Abundance, color = Week)) + 
  geom_boxplot() + geom_point() +  
  labs(y="Relative Abundance", title = str_glue(ancom.intersect.hfd$taxon[1], " Relative Abundance, fecal microbiome"),
       x="Surgery") + 
  theme_bw(base_size = 14) + 
  facet_wrap(~Surgery)
```

```{r}
fecal.graph.hfd %>% 
  filter(OTU == "Lactobacillaceae") %>% 
  group_by(Week) %>%
  kruskal_test(Abundance ~ Surgery) %>%
  adjust_pvalue(method = "holm") %>%
  add_significance()
```

Post-hoc: pairwise comparison with Wilcoxon test. 

```{r}
fecal.graph.hfd %>% 
  filter(OTU == "Lactobacillaceae") %>% 
  group_by(Week) %>% 
  wilcox_test(Abundance ~ Surgery, p.adjust.method = "holm") %>%
  add_significance() %>% 
  add_y_position() -> fecal.hfd.res

fecal.hfd.res
```



```{r, fig.width=6, fig.height=5}
week.labs <- c("Week 0", "Week 10", "Week 13")
names(week.labs) <- c("0", "10", "13")

fecal.graph.hfd %>% 
  filter(OTU == "Lactobacillaceae") %>% 
  ggplot(aes(x = Surgery, y = Abundance, color = Surgery)) + 
  geom_boxplot() + geom_point() +  
  labs(y="Lactobacillaceae Relative Abundance",
       x="Surgery") + 
  theme_bw(base_size = 14) + 
  facet_wrap(~Week,  labeller = labeller(Week = week.labs)) + theme(legend.position = "top") -> fecal.hfd.res.graph

fecal.hfd.res.graph
```


```{r, fig.width=7, fig.height=5}
fecal.hfd.res$y.position <- c(0.45, 0.5, 0.57, 0.55, 0.64, 0.7)
fecal.hfd.res.graph  + 
  stat_pvalue_manual(fecal.hfd.res, label = "p.adj.signif") -> fecal.hfd.res.graph

fecal.hfd.res.graph
```


```{r}
fecal.graph.hfd %>% 
  filter(OTU %in% ancom.intersect.hfd$taxon) %>%   
  ggplot(aes(x = Surgery, y = Abundance, color = Surgery)) + 
  geom_boxplot() + geom_point() +  
  labs(y="Relative Abundance", title = str_glue(ancom.intersect.hfd$taxon[2], " Relative Abundance, fecal microbiome"),
       x="Surgery") + 
  theme_bw(base_size = 14) + 
  facet_wrap(~Week)

```

```{r}
fecal.graph.hfd %>% 
  filter(OTU %in% ancom.intersect.hfd$taxon) %>%   
  ggplot(aes(x = Week, y = Abundance, color = Week)) + 
  geom_boxplot() + geom_point() +  
  labs(y="Relative Abundance", title = str_glue(ancom.intersect.hfd$taxon[2], " Relative Abundance, fecal microbiome"),
       x="Surgery") + 
  theme_bw(base_size = 14) + 
  facet_wrap(~Surgery)
```

```{r}
fecal.graph.hfd %>% 
  filter(OTU == "Lactobacillales") %>% 
  group_by(Week) %>%
  kruskal_test(Abundance ~ Surgery) %>%
  adjust_pvalue(method = "holm") %>%
  add_significance()
```

Post-hoc: pairwise comparison with Wilcoxon test. 

```{r}
fecal.graph.hfd %>% 
  filter(OTU == "Lactobacillales") %>% 
  group_by(Week) %>% 
  wilcox_test(Abundance ~ Surgery, p.adjust.method = "holm") %>%
  add_significance() %>% 
  add_y_position() -> fecal.hfd.res.order

fecal.hfd.res.order
```


```{r, fig.width=6, fig.height=5}
fecal.graph.hfd %>% 
  filter(OTU == "Lactobacillales") %>% 
  ggplot(aes(x = Surgery, y = Abundance, color = Surgery)) + 
  geom_boxplot() + geom_point() +  
  labs(y="Lactobacillales Relative Abundance",
       x="Surgery") + 
  theme_bw(base_size = 14) + 
  facet_wrap(~Week,  labeller = labeller(Week = week.labs)) + theme(legend.position = "top") -> fecal.hfd.res.graph.order

fecal.hfd.res.graph.order
```


```{r, fig.width=7, fig.height=5}
fecal.hfd.res.order$y.position <- c(0.45, 0.5, 0.57, 0.64, 0.7, 0.76)
fecal.hfd.res.graph.order  + 
  stat_pvalue_manual(fecal.hfd.res.order, label = "p.adj.signif") -> fecal.hfd.res.graph.order

fecal.hfd.res.graph.order
```

```{r, fig.width=10, fig.height=5}
plot_grid(fecal.hfd.res.graph, fecal.hfd.res.graph.order, 
          labels = "AUTO", ncol = 2, label_size = 20, scale = 0.9) -> fecal.hfd.summary
```

```{r}
ragg::agg_jpeg(file.path(fig.dir, "fecal_surgery_lacto_summary.jpeg"), width = 10, height = 5, units = "in", res = 600)
fecal.hfd.summary
dev.off()
```


# Q2: no surgery group, all weeks, Diet * Week + Genotype

Let's see how many mice we have: 

```{r}
meta.fecal %>% filter(Surgery == "None") -> meta.f.nosurgery
meta.f.nosurgery %>% 
  dplyr::count(Week, Diet, Genotype) %>% 
  arrange(n)
```


```{r}
meta.f.nosurgery %>% 
  dplyr::count(Week, Diet) %>% 
  arrange(n)
```

I will run Diet * Week + Genotype. 

```{r}
meta.f.nosurgery %>% distinct(Mouse) %>% nrow()
```

## Make TSE objects

```{r}
ps_genus.prune %>% subset_samples(Surgery == "None")  %>%
  mia::makeTreeSummarizedExperimentFromPhyloseq() -> tse.fecal.Q2.Genus
tse.fecal.Q2.Genus
```

```{r}
ps_Family.prune %>% subset_samples(Surgery == "None")  %>%
  mia::makeTreeSummarizedExperimentFromPhyloseq() -> tse.fecal.Q2.Family
tse.fecal.Q2.Family

ps_Order.prune %>% subset_samples(Surgery == "None")  %>%
  mia::makeTreeSummarizedExperimentFromPhyloseq() -> tse.fecal.Q2.Order
tse.fecal.Q2.Order

ps_Class.prune %>% subset_samples(Surgery == "None")  %>%
  mia::makeTreeSummarizedExperimentFromPhyloseq() -> tse.fecal.Q2.Class
tse.fecal.Q2.Class

ps_Phylum.prune %>% subset_samples(Surgery == "None")  %>%
  mia::makeTreeSummarizedExperimentFromPhyloseq() -> tse.fecal.Q2.Phylum
tse.fecal.Q2.Phylum
```

## ANCOM II 

###  Genus

```{r}
set.seed(123) # Set seed for reproducibility 

# Genus level
fecal.Q2.Genus <- ANCOMBC::ancom(data=tse.fecal.Q2.Genus, assay_name="counts", tax_level="Genus",
                               p_adj_method="holm", prv_cut=0, lib_cut=0, 
                               main_var="Diet", adj_formula="Week + Genotype",
                               alpha = 0.05, struc_zero=TRUE,
                               n_cl = n_cl_requested) 

res.fecal.Q2.Genus = fecal.Q2.Genus$res
res.fecal.Q2.Genus %>% filter(detected_0.7 == TRUE) -> tab.fecal.Q2.Genus.sig
tab.fecal.Q2.Genus.sig 
```

### Family

```{r}
set.seed(123) # Set seed for reproducibility 

# Family level
fecal.Q2.Family <- ANCOMBC::ancom(data=tse.fecal.Q2.Family, assay_name="counts", tax_level="Family",
                               p_adj_method="holm", prv_cut=0, lib_cut=0, 
                               main_var="Diet", adj_formula="Week + Genotype",
                               alpha = 0.05, struc_zero=TRUE,
                               n_cl = n_cl_requested) 

res.fecal.Q2.Family = fecal.Q2.Family$res
res.fecal.Q2.Family %>% filter(detected_0.7 == TRUE) -> tab.fecal.Q2.Family.sig
tab.fecal.Q2.Family.sig 
```

### Order

```{r}
set.seed(123) # Set seed for reproducibility 

# Order level
fecal.Q2.Order <- ANCOMBC::ancom(data=tse.fecal.Q2.Order, assay_name="counts", tax_level="Order",
                               p_adj_method="holm", prv_cut=0, lib_cut=0, 
                               main_var="Diet", adj_formula="Week + Genotype",
                               alpha = 0.05, struc_zero=TRUE,
                               n_cl = n_cl_requested) 

res.fecal.Q2.Order = fecal.Q2.Order$res
res.fecal.Q2.Order %>% filter(detected_0.7 == TRUE) -> tab.fecal.Q2.Order.sig
tab.fecal.Q2.Order.sig 
```

### Class

```{r}
set.seed(123) # Set seed for reproducibility 

# Class level
fecal.Q2.Class <- ANCOMBC::ancom(data=tse.fecal.Q2.Class, assay_name="counts", tax_level="Class",
                               p_adj_method="holm", prv_cut=0, lib_cut=0, 
                               main_var="Diet", adj_formula="Week + Genotype",
                               alpha = 0.05, struc_zero=TRUE,
                               n_cl = n_cl_requested) 

res.fecal.Q2.Class = fecal.Q2.Class$res
res.fecal.Q2.Class %>% filter(detected_0.7 == TRUE) -> tab.fecal.Q2.Class.sig
tab.fecal.Q2.Class.sig 
```

### Phylum

```{r}
set.seed(123) # Set seed for reproducibility 

# Phylum level
fecal.Q2.Phylum <- ANCOMBC::ancom(data=tse.fecal.Q2.Phylum, assay_name="counts", tax_level="Phylum",
                               p_adj_method="holm", prv_cut=0, lib_cut=0, 
                               main_var="Diet", adj_formula="Week + Genotype",
                               alpha = 0.05, struc_zero=TRUE,
                               n_cl = n_cl_requested) 

res.fecal.Q2.Phylum = fecal.Q2.Phylum$res
res.fecal.Q2.Phylum %>% filter(detected_0.7 == TRUE) -> tab.fecal.Q2.Phylum.sig
tab.fecal.Q2.Phylum.sig 
```


### Aggregate Results

```{r}
tab.fecal.Q2.Genus.sig$Tax_rank <- "Genus"
tab.fecal.Q2.Family.sig$Tax_rank <- "Family"
tab.fecal.Q2.Order.sig$Tax_rank <- "Order"
tab.fecal.Q2.Class.sig$Tax_rank <- "Class"
tab.fecal.Q2.Phylum.sig$Tax_rank <- "Phylum"

rbind(tab.fecal.Q2.Genus.sig,
      tab.fecal.Q2.Family.sig,
      tab.fecal.Q2.Order.sig,
      tab.fecal.Q2.Class.sig,
      tab.fecal.Q2.Phylum.sig) -> tab.fecal.Q2

tab.fecal.Q2$Test <- "ANCOM II, fecal microbiome: no surgery, main effect Diet (structural zeroes) and week + genotype"

tab.fecal.Q2 
```

## DESeq2

### Genus 

```{r}
ds.fecal.Q2_Genus <- DESeq2::DESeqDataSet(tse.fecal.Q2.Genus, ~ Diet * Week + Genotype)
dds.fecal.Q2_Genus <- DESeq2::DESeq(ds.fecal.Q2_Genus)

DESeq2::summary(dds.fecal.Q2_Genus)
resultsNames(dds.fecal.Q2_Genus)
```

```{r}
results(dds.fecal.Q2_Genus, tidy = TRUE, name =c("Diet_High_Fat_Diet_vs_Normal_Chow")) %>% # tidy converts to df
  filter(padj < 0.05) 

results(dds.fecal.Q2_Genus, name =c("Genotype_KO_vs_WT"), tidy = TRUE) %>%
  filter(padj < 0.05) -> fecal.Q2.Genus.Res
fecal.Q2.Genus.Res$Comparison <- c("Genotype: KO vs. WT")

results(dds.fecal.Q2_Genus, name =c("DietHigh_Fat_Diet.Week10"), tidy = TRUE) %>%
  filter(padj < 0.05) -> tmp
tmp$Comparison <- c("HFD, wk 10")
rbind(tmp, fecal.Q2.Genus.Res) -> fecal.Q2.Genus.Res

results(dds.fecal.Q2_Genus, name =c("DietHigh_Fat_Diet.Week13"), tidy = TRUE) %>%
  filter(padj < 0.05) -> tmp
tmp$Comparison <- c("HFD, wk 13")
rbind(tmp, fecal.Q2.Genus.Res) -> fecal.Q2.Genus.Res

results(dds.fecal.Q2_Genus, name =c("Week_10_vs_0"), tidy = TRUE) %>%
  filter(padj < 0.05) -> tmp
tmp$Comparison <- c("Week: 10 vs. 0")
rbind(tmp, fecal.Q2.Genus.Res) -> fecal.Q2.Genus.Res

results(dds.fecal.Q2_Genus, name =c("Week_13_vs_0"), tidy = TRUE) %>%
  filter(padj < 0.05) -> tmp
tmp$Comparison <- c("Week: 13 vs. 0")
rbind(tmp, fecal.Q2.Genus.Res) -> fecal.Q2.Genus.Res

fecal.Q2.Genus.Res$Compare_taxon <- "Genus"

fecal.Q2.Genus.Res
```

### Family 

```{r}
ds.fecal.Q2_Family <- DESeq2::DESeqDataSet(tse.fecal.Q2.Family, ~ Diet * Week + Genotype)
dds.fecal.Q2_Family <- DESeq2::DESeq(ds.fecal.Q2_Family)

DESeq2::summary(dds.fecal.Q2_Family)
```

```{r}
results(dds.fecal.Q2_Family, tidy = TRUE, name =c("Diet_High_Fat_Diet_vs_Normal_Chow")) %>% # tidy converts to df
  filter(padj < 0.05) -> fecal.Q2.Family.Res
fecal.Q2.Family.Res$Comparison <- c("Diet: HFD vs. NC")

results(dds.fecal.Q2_Family, name =c("Genotype_KO_vs_WT"), tidy = TRUE) %>%
  filter(padj < 0.05) -> tmp
tmp$Comparison <- c("Genotype: KO vs. WT")
rbind(tmp, fecal.Q2.Family.Res) -> fecal.Q2.Family.Res

results(dds.fecal.Q2_Family, name =c("DietHigh_Fat_Diet.Week10"), tidy = TRUE) %>%
  filter(padj < 0.05) -> tmp
tmp$Comparison <- c("HFD, wk 10")
rbind(tmp, fecal.Q2.Family.Res) -> fecal.Q2.Family.Res

results(dds.fecal.Q2_Family, name =c("DietHigh_Fat_Diet.Week13"), tidy = TRUE) %>%
  filter(padj < 0.05) -> tmp
tmp$Comparison <- c("HFD, wk 13")
rbind(tmp, fecal.Q2.Family.Res) -> fecal.Q2.Family.Res

results(dds.fecal.Q2_Family, name =c("Week_10_vs_0"), tidy = TRUE) %>%
  filter(padj < 0.05) -> tmp
tmp$Comparison <- c("Week: 10 vs. 0")
rbind(tmp, fecal.Q2.Family.Res) -> fecal.Q2.Family.Res

results(dds.fecal.Q2_Family, name =c("Week_13_vs_0"), tidy = TRUE) %>%
  filter(padj < 0.05) -> tmp
tmp$Comparison <- c("Week: 13 vs. 0")
rbind(tmp, fecal.Q2.Family.Res) -> fecal.Q2.Family.Res

fecal.Q2.Family.Res$Compare_taxon <- "Family"

fecal.Q2.Family.Res
```

### Order 

```{r}
ds.fecal.Q2_Order <- DESeq2::DESeqDataSet(tse.fecal.Q2.Order, ~ Diet * Week + Genotype)
dds.fecal.Q2_Order <- DESeq2::DESeq(ds.fecal.Q2_Order)

DESeq2::summary(dds.fecal.Q2_Order)
```

```{r}
results(dds.fecal.Q2_Order, tidy = TRUE, name =c("Diet_High_Fat_Diet_vs_Normal_Chow")) %>% # tidy converts to df
  filter(padj < 0.05) 

results(dds.fecal.Q2_Order, name =c("Genotype_KO_vs_WT"), tidy = TRUE) %>%
  filter(padj < 0.05) -> fecal.Q2.Order.Res
fecal.Q2.Order.Res$Comparison <- c("Genotype: KO vs. WT")

results(dds.fecal.Q2_Order, name =c("DietHigh_Fat_Diet.Week10"), tidy = TRUE) %>%
  filter(padj < 0.05) -> tmp
tmp$Comparison <- c("HFD, wk 10")
rbind(tmp, fecal.Q2.Order.Res) -> fecal.Q2.Order.Res

results(dds.fecal.Q2_Order, name =c("DietHigh_Fat_Diet.Week13"), tidy = TRUE) %>%
  filter(padj < 0.05) -> tmp
tmp$Comparison <- c("HFD, wk 13")
rbind(tmp, fecal.Q2.Order.Res) -> fecal.Q2.Order.Res

results(dds.fecal.Q2_Order, name =c("Week_10_vs_0"), tidy = TRUE) %>%
  filter(padj < 0.05) 

results(dds.fecal.Q2_Order, name =c("Week_13_vs_0"), tidy = TRUE) %>%
  filter(padj < 0.05) -> tmp
tmp$Comparison <- c("Week: 13 vs. 0")
rbind(tmp, fecal.Q2.Order.Res) -> fecal.Q2.Order.Res

fecal.Q2.Order.Res$Compare_taxon <- "Order"

fecal.Q2.Order.Res
```
	
	
### Class 

```{r}
ds.fecal.Q2_Class <- DESeq2::DESeqDataSet(tse.fecal.Q2.Class, ~ Diet * Week + Genotype)
dds.fecal.Q2_Class <- DESeq2::DESeq(ds.fecal.Q2_Class)

DESeq2::summary(dds.fecal.Q2_Class)
```

```{r}
results(dds.fecal.Q2_Class, tidy = TRUE, name =c("Diet_High_Fat_Diet_vs_Normal_Chow")) %>% # tidy converts to df
  filter(padj < 0.05) 

results(dds.fecal.Q2_Class, name =c("Genotype_KO_vs_WT"), tidy = TRUE) %>%
  filter(padj < 0.05) -> fecal.Q2.Class.Res
fecal.Q2.Class.Res$Comparison <- c("Genotype: KO vs. WT")

results(dds.fecal.Q2_Class, name =c("DietHigh_Fat_Diet.Week10"), tidy = TRUE) %>%
  filter(padj < 0.05) -> tmp
tmp$Comparison <- c("HFD, wk 10")
rbind(tmp, fecal.Q2.Class.Res) -> fecal.Q2.Class.Res

results(dds.fecal.Q2_Class, name =c("DietHigh_Fat_Diet.Week13"), tidy = TRUE) %>%
  filter(padj < 0.05) -> tmp
tmp$Comparison <- c("HFD, wk 13")
rbind(tmp, fecal.Q2.Class.Res) -> fecal.Q2.Class.Res

results(dds.fecal.Q2_Class, name =c("Week_10_vs_0"), tidy = TRUE) %>%
  filter(padj < 0.05) 

results(dds.fecal.Q2_Class, name =c("Week_13_vs_0"), tidy = TRUE) %>%
  filter(padj < 0.05) -> tmp
tmp$Comparison <- c("Week: 13 vs. 0")
rbind(tmp, fecal.Q2.Class.Res) -> fecal.Q2.Class.Res

fecal.Q2.Class.Res$Compare_taxon <- "Class"

fecal.Q2.Class.Res
```
	
### Phylum 

```{r}
ds.fecal.Q2_Phylum <- DESeq2::DESeqDataSet(tse.fecal.Q2.Phylum, ~ Diet * Week + Genotype)
dds.fecal.Q2_Phylum <- DESeq2::DESeq(ds.fecal.Q2_Phylum)

DESeq2::summary(dds.fecal.Q2_Phylum)
```

```{r}
results(dds.fecal.Q2_Phylum, tidy = TRUE, name =c("Diet_High_Fat_Diet_vs_Normal_Chow")) %>% # tidy converts to df
  filter(padj < 0.05) 

results(dds.fecal.Q2_Phylum, name =c("Genotype_KO_vs_WT"), tidy = TRUE) %>%
  filter(padj < 0.05) -> fecal.Q2.Phylum.Res
fecal.Q2.Phylum.Res$Comparison <- c("Genotype: KO vs. WT")

results(dds.fecal.Q2_Phylum, name =c("DietHigh_Fat_Diet.Week10"), tidy = TRUE) %>%
  filter(padj < 0.05) -> tmp
tmp$Comparison <- c("HFD, wk 10")
rbind(tmp, fecal.Q2.Phylum.Res) -> fecal.Q2.Phylum.Res

results(dds.fecal.Q2_Phylum, name =c("DietHigh_Fat_Diet.Week13"), tidy = TRUE) %>%
  filter(padj < 0.05) -> tmp
tmp$Comparison <- c("HFD, wk 13")
rbind(tmp, fecal.Q2.Phylum.Res) -> fecal.Q2.Phylum.Res

results(dds.fecal.Q2_Phylum, name =c("Week_10_vs_0"), tidy = TRUE) %>%
  filter(padj < 0.05) 

results(dds.fecal.Q2_Phylum, name =c("Week_13_vs_0"), tidy = TRUE) %>%
  filter(padj < 0.05) -> tmp
tmp$Comparison <- c("Week: 13 vs. 0")
rbind(tmp, fecal.Q2.Phylum.Res) -> fecal.Q2.Phylum.Res

fecal.Q2.Phylum.Res$Compare_taxon <- "Phylum"

fecal.Q2.Phylum.Res
```
	
### Aggregate results

```{r}
rbind(fecal.Q2.Genus.Res, 
      fecal.Q2.Family.Res,
      fecal.Q2.Class.Res,
      fecal.Q2.Order.Res,
      fecal.Q2.Phylum.Res) -> fecal.Q2.deseq

fecal.Q2.deseq$Test <- "DESeq2, Fecal microbiome: no surgery, ~ Diet * Week + Genotype"

fecal.Q2.deseq
```

## Intersection 

```{r}
tab.fecal.Q2 %>% filter(taxon %in% fecal.Q2.deseq$row) -> ancom.intersect.nosurgery
ancom.intersect.nosurgery
```

```{r}
fecal.Q2.deseq %>% filter(row %in% tab.fecal.Q2$taxon) %>%
  dplyr::select(row, Comparison, Compare_taxon, Test)  %>%
  group_by(row) %>%
  mutate(n.ind = paste0(1:n()))  %>%
  pivot_wider(
    names_from = "n.ind",
    values_from = Comparison
  )  -> fecal.Q2.deseq.intersect.wide

fecal.Q2.deseq.intersect.wide
```

## Graph

```{r}
fecal.RA.Full %>% filter(Surgery == "None") -> fecal.graph.nosurgery
head(fecal.graph.nosurgery)
tail(fecal.graph.nosurgery)
```

```{r, fig.width=6, fig.height=5}
for (i in 1:length(ancom.intersect.nosurgery$taxon)) {
  fecal.graph.nosurgery %>% 
  filter(OTU == ancom.intersect.nosurgery$taxon[i]) %>%  
  ggplot(aes(x = as.character(Week), y = Abundance, color = Diet)) + 
  geom_boxplot() + geom_point() +  
  labs(y="Relative Abundance", title = str_glue(ancom.intersect.nosurgery$taxon[i], " Relative Abundance, fecal"),
       x="Week") + 
  facet_wrap(~Diet) + 
  theme_bw(base_size = 14) + 
  theme(legend.position = "top") -> p
  print(p)
}
```

## Graph: by diet 

```{r, fig.width=8, fig.height=5}
for (i in 1:length(ancom.intersect.nosurgery$taxon)) {
  fecal.graph.nosurgery %>% 
  filter(OTU == ancom.intersect.nosurgery$taxon[i]) %>%  
  ggplot(aes(x = Diet, y = Abundance, color = Diet)) + 
  geom_boxplot() + geom_point() +  
  labs(y="Relative Abundance", title = str_glue(ancom.intersect.nosurgery$taxon[i], " Relative Abundance, fecal"),
       x="Diet") + 
  facet_wrap(~Week) + 
  theme_bw(base_size = 14) + 
  theme(legend.position = "top") + 
  stat_compare_means(method = "wilcox.test", aes(label = after_stat(p.signif)),
                     label.x = 1.5) -> p
  print(p)
}
```


# Genotype? 

DESeq2 provides more information on the results than ANCOM II does in the sense that you can specifically set the contrast. There were a number of taxa that were differentially abundant based on genotype: 

```{r}
fecal.Q2.deseq %>% filter(row %in% tab.fecal.Q2$taxon) %>% 
  filter(str_detect(Comparison, "Genotype") == TRUE) -> nosurgery.genotype

nosurgery.genotype
```

```{r}
fecal.Q2.deseq.intersect.wide %>% filter(row %in% nosurgery.genotype$row)
```

It looks like these taxa were also affected by diet and week. 

```{r, fig.width=8, fig.height=5}
for (i in 1:length(nosurgery.genotype$row)) {
  fecal.graph.nosurgery %>% 
  filter(OTU == nosurgery.genotype$row[i]) %>%  
  ggplot(aes(x = Genotype, y = Abundance, color = Genotype)) + 
  geom_boxplot() + geom_point() +  
  labs(y="Relative Abundance", title = str_glue(nosurgery.genotype$row[i], " Relative Abundance, fecal"),
       x="Diet") + 
  facet_grid(vars(Diet), vars(Week)) + 
  theme_bw(base_size = 14)-> p
  print(p)
}
```

```{r}
fecal.graph.nosurgery %>% 
  filter(OTU %in% nosurgery.genotype$row) %>%
  group_by(OTU, Diet, Week) %>% 
  filter(sum(Abundance) > 0) %>% # remove groups that are all zeroes 
  wilcox_test(Abundance ~ Genotype) %>%
  adjust_pvalue(method = "holm") %>%
  add_significance()
```

NS after multiple testing. 

# Export csv files 

```{r}
fecal.Q1.deseq %>% filter(row %in% ancom.intersect.hfd$taxon) -> fecal.Q1.deseq.intersect
fecal.Q2.deseq %>% filter(row %in% ancom.intersect.nosurgery$taxon) -> fecal.Q2.deseq.intersect
```

```{r}
write_csv(fecal.RA.Full, file.path(git.dir, "fecal_relative_abundance.csv"), append = FALSE, col_names = TRUE)
write_csv(ancom.intersect.hfd, file.path(git.dir, "fecal_ancom_hfd_intersect.csv"), append = FALSE, col_names = TRUE)
write_csv(ancom.intersect.nosurgery, file.path(git.dir, "fecal_ancom_nosurgery_intersect.csv"), append = FALSE, col_names = TRUE)
write_csv(fecal.Q1.deseq.intersect, file.path(git.dir, "fecal_deseq_hfd_intersect.csv"), append = FALSE, col_names = TRUE)
write_csv(fecal.Q2.deseq.intersect, file.path(git.dir, "fecal_deseq_nosurgery_intersect.csv"), append = FALSE, col_names = TRUE)
```




# Reproducibility
```{r}
sessionInfo()
```

